modules/navier_stokes/src/fv/kernels/INSFVMomentumPressureRZ.C

```diff
@@ -0,0 +1,40 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVMomentumPressureRZ.h"
+
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", INSFVMomentumPressureRZ);
+
+InputParameters
+INSFVMomentumPressureRZ::validParams()
+{
+  InputParameters params = FVElementalKernel::validParams();
+  params.addClassDescription("Adds the $-p/r$ term into the radial component of the Navier-Stokes "
+                             "momentum equation for the problems in the RZ coordinate system.");
+  params.addRequiredCoupledVar("p", "The pressure");
+  return params;
+}
+
+INSFVMomentumPressureRZ::INSFVMomentumPressureRZ(const InputParameters & params)
+  : FVElementalKernel(params), _p(adCoupledValue("p"))
+{
+}
+
+ADReal
+INSFVMomentumPressureRZ::computeQpResidual()
+{
+  mooseAssert(_subproblem.getCoordSystem(_current_elem->subdomain_id()) == Moose::COORD_RZ,
+              "This object should only be active in an RZ coordinate system.");
+
+  auto rz_radial_coord = _subproblem.getAxisymmetricRadialCoord();
+
+  return -_p[_qp] / _q_point[_qp](rz_radial_coord);
```

Comment:
So is there a fix already for FVElementalKernels ? #15194

Reply:
No. This would fail if you had any variables in your nonlinear system with a basis other than `CONSTANT MONOMIAL`

Reply:
ok so we're good for now though. It's just materials, I misread the issue.

Reply:
If the quadrature rule is wrong for materials, then it will be wrong for FVKernels. I don't think @rwcarlsen's wording there is 100% clear. What I think he means is that *if* FE and FV variables exist in a single nonlinear system, then material properties that will be used by finite volume objects will be initialized with the wrong quadrature rule. Additionally, *if* FE and FV variables exist in the same nonlinear system, then the above `_q_point` would also be wrong.

Reply:
I am pretty confident that if only FV variables exist, that materials will be initialized and evaluated with a single point quadrature rule